### DISCUSSION TITLE
failed to create nodeimage

### DISCUSSION THREAD

**Discussion by haozi4go** [openkruise/kruise#1080] [Q&A]
部署CloneSet报如下错误是啥原因呢？
W0920 11:29:44.146888       1 nodeimage_controller.go:154] Failed to process NodeImage master01 err 30.003071573s, elapsedTime failed to create nodeimage master01, err: Internal error occurred: failed calling webhook "mnodeimage.kb.io": Post "https://kruise-webhook-service.kruise-system.svc:443/mutate-apps-kruise-io-v1alpha1-nodeimage?timeout=30s": context deadline exceeded
E0920 11:29:44.146967       1 controller.go:317] controller/nodeimage-controller "msg"="Reconciler error" "error"="failed to create nodeimage master01, err: Internal error occurred: failed calling webhook \"mnodeimage.kb.io\": Post \"https://kruise-webhook-service.kruise-system.svc:443/mutate-apps-kruise-io-v1alpha1-nodeimage?timeout=30s\": context deadline exceeded" "name"="master01" "namespace"=""

**Comment by FillZpp** [comment]
这个是首次部署 kruise 短暂出现的，还是持续在报错？

如果是持续出现，你要先检查下在 kube-apiserver 容器中能否连通 kruise-controller-manager 所在的网络地址。

**Reply by haozi4go** [reply]
是持续出现的，发现kruise-webhook-service.kruise-system.svc域名解析出来的ip不在pod的网段，这个ip网段可以配置或指定吗

**Reply by FillZpp** [reply]
Pod IP 是你们集群中 CNI 组件分配的，这个要看你们自己的网络相关配置了
